File: vuelos_frontend/src/core/services/reserva.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Reserva, ReservaCreate } from '../models/reserva.model';
import { environment } from '../../environments/environment';

@Injectable({ providedIn: 'root' })
export class ReservaService {
  private http = inject(HttpClient);
  private base = `${environment.apiUrl}/reservas/`;

  listar(): Observable<Reserva[]> {
    return this.http.get<Reserva[]>(this.base);
  }

  crear(payload: ReservaCreate): Observable<Reserva> {
    return this.http.post<Reserva>(this.base, payload);
  }

  buscar(documento: string, numeroVuelo: string): Observable<Reserva> {
    const params = new HttpParams()
      .set('documento', documento)
      .set('numero_vuelo', numeroVuelo);
    return this.http.get<Reserva>(`${this.base}buscar/`, { params });
  }

  buscarPost(documento: string, numeroVuelo: string): Observable<Reserva> {
    return this.http.post<Reserva>(`${this.base}buscar-post/`, {
      documento,
      numero_vuelo: numeroVuelo,
    });
  }
}

File: vuelos_api_java/src/main/java/pe/edu/upeu/vuelos_api/service/ReservaService.java
package pe.edu.upeu.vuelos_api.service;

import pe.edu.upeu.vuelos_api.dto.ReservaCreateDTO;
import pe.edu.upeu.vuelos_api.dto.ReservaDTO;

import java.util.List;

public interface ReservaService {

    List<ReservaDTO> listar();

    ReservaDTO crear(ReservaCreateDTO dto);

    ReservaDTO buscar(String documento, String numeroVuelo);
}
